File: AppBanNoiThat/app/src/main/java/com/example/appbannoithat/Model/NguoiDung.kt
package com.example.appbannoithat.Model


data class Account (
    val _id: String,
    val ten_tai_khoan: String,
    val email: String,
    val mat_khau: String,
    val ho_ten: String,
    val dia_chi: String,
    val so_dien_thoai: String,
    val role: Boolean,
    val socketId: String
)

data class NguoiDungDK (
    val ten_tai_khoan: String,
    val email: String,
    val mat_khau: String,
    val ho_ten: String,
    val role: Boolean,
    val socketId: String
)

data class NguoiDungDN (
    val ten_tai_khoan: String,
    val mat_khau: String
)

data class updateSocket (
    val id: String,
    val socket: String
)

File: API_AppBanNoiThat/model/nguoiDung.js
const mongoose = require('mongoose')

const NguoiDungsSchema = new mongoose.Schema({
    ten_tai_khoan:{
        type: String,
        minlength: 10,
        maxlength: 15,
        unique: true
    },
    email: {
        type: String,
        unique: true
    },
    mat_khau: {
        type: String,
        minlength: 8,
        maxlength: 15,
        require: true,
    },
    ho_ten: {
        type: String
    },
    dia_chi: {
        type: String
    },
    so_dien_thoai: {
        type: String
    },
    role:{
        type: Boolean,
        default: false
    },
    socketId:{
        type: String
    }
},{
    timestamps: true
})
module.exports = new mongoose.model('nguoiDungs', NguoiDungsSchema)